npm/extensions.bzl

```diff
@@ -148,7 +126,7 @@ WARNING: Cannot determine home directory in order to load home `.npmrc` file in
     imports = npm_translate_lock_helpers.get_npm_imports(
         importers = importers,
         packages = packages,
-        patched_dependencies = lock_patched_dependencies,
+        patched_dependencies = state.patched_dependencies(),
```

Comment:
The goal in 2.x is adding:
```
only_built_dependencies = state.only_built_dependencies()`
```

This PR is to allow the parsing of the root package.json to remain only in the `state` util and not bleed into this bzlmod specific file.